rfcs/build-system/04-build-shield.md

```diff
@@ -0,0 +1,107 @@
+
+Some weeks you'll get lucky and have almost nothing to do, but other weeks something will "blow up" in a way that requires much more time to address. If the time goes beyond ~25% _(or whatever number we decide above)_ it's worth calling out to management and/or pulling in other people as needed.
+
+### Responsibilities
+
+Starting out, the responsibility of build shield is limited to ensuring that PR/CI/release builds and local development stay unblocked (and documenting any failures that occur).
+
+#### Keeping builds unblocked
+
+This includes monitoring for:
+
+- Automated failure notifications in "Build failures (automated)" channel
+- Messages from team members (mainly in "Engineering Systems" channel) about PR or local build issues
+- Messages or github mentions from regular shield about urgent build-related regressions/breaks that showed up as github issues
+  - This is much less common and will primarily be up to the regular shield PM/dev to watch for these issues and loop in build shield
+
+Prioritization is roughly as follows. As with normal shield, it's fine to pull in others for consultation/help as needed, especially if it's a particularly complex or urgent issue, or if a more in-depth fix turns out to be necessary.
+
+| Issue type                                      | Priority                                                                     |
+| ----------------------------------------------- | ---------------------------------------------------------------------------- |
+| Release build failed                            | 🔥 Investigate ASAP (fix issue, or retry if intermittent)                    |
+| CI _consistently_ broken                        | 🔥 Fix ASAP                                                                  |
+| PR builds _consistently_ broken (CI okay)       | 🔥 Fix ASAP                                                                  |
+| Local builds _consistently_ broken (PR/CI okay) | 🔥 Fix ASAP                                                                  |
+| Published package broken                        | 🔥 Fix ASAP (usually due to dep issue or missing file)                       |
+| CI _intermittently_ broken                      | ⚠️ Try to investigate/fix, or file an issue if it will take significant time |
+| v7/8 website release failed (uncommon)          | ⚠️ Re-run failed stage, or see internal wiki                                 |
+
+#### Documenting issues and troubleshooting procedures
+
+The second responsibility is **documenting** failures and troubleshooting procedures, with the goals of:
+
+- Noticing failure patterns which we need to schedule work to address
+- Establishing a "runbook" for troubleshooting build issues
+  - The goal of this is to de-mystify build, and maybe even (eventually) be able to expand the build shield rotation to more team members, so that everyone has exposure to the area
```

Comment:
This section is an important goal. Consistently using and improving a runbook will allow us to reach more self-serve on build issues that folks hit. Ideally this can also drive improvements or simplification into our build setup.